File: app/src/main/java/com/textfield/json/outabus/Stop.java
package com.textfield.json.outabus;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;

/**
 * Created by Jason on 19/04/2016.
 */
public class Stop implements Parcelable{
    private String name, id;
    private double lat, lng;
    public ArrayList<Bus> buses;

    public Stop(String id, String name, double lat, double lng, ArrayList<Bus> buses) {
        this.id = id;
        this.name = name;
        this.lat = lat;
        this.lng = lng;
        this.buses = buses;
    }

    public Stop(String id, String name, double lat, double lng) {
        this.id = id;
        this.name = name;
        this.lat = lat;
        this.lng = lng;
        this.buses = new ArrayList<>();
    }
    public Stop(Parcel in){
        double[] data = new double[2];
        in.readDoubleArray(data);
        lat = data[0];
        lng = data[1];
        name = in.readString();
    }

    public static final Parcelable.Creator CREATOR = new Parcelable.Creator() {
        public Stop createFromParcel(Parcel in) {
            return new Stop(in);
        }

        public Stop[] newArray(int size) {
            return new Stop[size];
        }
    };

    @Override
    public boolean equals(Object o) {
        return id.equals(((Stop) o).id);
    }

    @Override
    public int hashCode() {
        int res = 0;
        for (char x : id.toCharArray()) {
            res += x;
        }
        return res;
    }

    public double getLat() {
        return lat;
    }

    public double getLng() {
        return lng;
    }

    public String getId() {
        return id;
    }

    public ArrayList<Bus> getBuses() {
        return buses;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name + " " + id + " " + buses;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeDoubleArray(new double[]{lat,lng});
        dest.writeString(name);
    }
}

File: app/src/main/java/com/textfield/json/outabus/entities/Stop.kt
package com.textfield.json.outabus.entities

import android.os.Parcel
import android.os.Parcelable

import java.util.ArrayList

/**
 * Created by Jason on 19/04/2016.
 */
class Stop : Parcelable {
    var name = ""
    var id = ""
    var lat = 0.0
    var lng = 0.0
    var buses = ArrayList<Bus>()

    constructor(id: String, name: String, lat: Double, lng: Double, buses: ArrayList<Bus>) {
        this.id = id
        this.name = name
        this.lat = lat
        this.lng = lng
        this.buses = buses
    }

    constructor(id: String, name: String, lat: Double, lng: Double) {
        this.id = id
        this.name = name
        this.lat = lat
        this.lng = lng
        this.buses = ArrayList()
    }

    constructor(`in`: Parcel) {
        lat = `in`.readDouble()
        lng = `in`.readDouble()
        name = `in`.readString()
    }

    override fun equals(other: Any?): Boolean {
        return id == (other as Stop).id
    }

    override fun hashCode(): Int {
        var res = 0
        for (x in id.toCharArray()) {
            res += x.toInt()
        }
        return res
    }

    override fun toString(): String {
        return "$name $id $buses"
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeDouble(lat)
        dest.writeDouble(lng)
        dest.writeString(name)
    }

    companion object CREATOR : Parcelable.Creator<Stop> {
        override fun createFromParcel(`in`: Parcel): Stop {
            return Stop(`in`)
        }

        override fun newArray(size: Int): Array<Stop?> {
            return arrayOfNulls(size)
        }
    }
}
